File: livingrimoire start here/LivinGrimoire java/older livingrimoire version/old java_core/DISkillUtils.java
package com.yotamarker.lgkotlin1;

import java.util.ArrayList;

public class DISkillUtils {
    public Algorithm verbatimGorithm(AbsAlgPart itte) {
        // returns a simple algorithm containing 1 alg part
        String representation = "util";
        ArrayList<AbsAlgPart> algParts1 = new ArrayList<>();
        algParts1.add(itte);
        Algorithm algorithm = new Algorithm("util", representation, algParts1);
        return algorithm;
    }

    public Algorithm verbatimGorithm(String algMarker, AbsAlgPart itte) {
        // returns a simple algorithm for saying sent parameter
        String representation = "util";
        ArrayList<AbsAlgPart> algParts1 = new ArrayList<>();
        algParts1.add(itte);
        Algorithm algorithm = new Algorithm("util", representation, algParts1);
        return algorithm;
    }

    public Algorithm customizedVerbatimGorithm(String algMarker, AbsAlgPart itte) {
        // the most stable and advanced algorithm builder
        // returns a simple algorithm containing 1 alg part
        String representation = "r_" + algMarker;
        ArrayList<AbsAlgPart> algParts1 = new ArrayList<>();
        algParts1.add(itte);
        Algorithm algorithm = new Algorithm(algMarker, representation, algParts1);
        return algorithm;
    }

    public Algorithm customizedVerbatimGorithm(String algMarker, AbsAlgPart... itte) {
        // the most stable and advanced algorithm builder
        // returns a simple algorithm containing 1 alg part
        String representation = "r_" + algMarker;
        ArrayList<AbsAlgPart> algParts1 = new ArrayList<>();
        for (int i = 0; i < itte.length; i++) {
            algParts1.add(itte[i]);
        }
        Algorithm algorithm = new Algorithm(algMarker, representation, algParts1);
        return algorithm;
    }

    public Algorithm simpleVerbatimAlgorithm(String algMarker, String... sayThis) {
        // returns alg that says the word string (sayThis)
        return customizedVerbatimGorithm(algMarker, new APVerbatim(sayThis));
    }

    public Algorithm simpleCloudiandVerbatimAlgorithm(CldBool cldBool, String algMarker, String... sayThis) {
        // returns alg that says the word string (sayThis)
        return customizedVerbatimGorithm(algMarker, new APCldVerbatim(cldBool, sayThis));
    }
    public String strContainsList(String str1, ArrayList<String> items) {
        // returns the 1st match between words in a string and values in a list.
        for (String temp : items) {
            if (str1.contains(temp)) {
                return temp;
            }
        }
        return "";
    }
}

File: livingrimoire start here/LivingGrimoire Kotlin/old LG kotlin version/kotlin_core/DISkillUtils.kt
package com.yotamarker.lgkotlinfull.LGCore

import java.util.*

class DISkillUtils {
    fun verbatimGorithm(itte: AbsAlgPart): Algorithm {
        // returns a simple algorithm containing 1 alg part
        val representation = "util"
        val algParts1 = ArrayList<AbsAlgPart>()
        algParts1.add(itte)
        return Algorithm("util", representation, algParts1)
    }

    fun verbatimGorithm(algMarker: String, itte: AbsAlgPart): Algorithm {
        // returns a simple algorithm for saying sent parameter
        val representation = "util"
        val algParts1 = ArrayList<AbsAlgPart>()
        algParts1.add(itte)
        return Algorithm("util", representation, algParts1)
    }

    fun customizedVerbatimGorithm(algMarker: String, itte: AbsAlgPart): Algorithm {
        // the most stable and advanced algorithm builder
        // returns a simple algorithm containing 1 alg part
        val representation = "r_$algMarker"
        val algParts1 = ArrayList<AbsAlgPart>()
        algParts1.add(itte)
        return Algorithm(algMarker, representation, algParts1)
    }

    fun customizedVerbatimGorithm(algMarker: String, vararg itte: AbsAlgPart): Algorithm {
        // the most stable and advanced algorithm builder
        // returns a simple algorithm containing 1 alg part
        val representation = "r_$algMarker"
        val algParts1 = ArrayList<AbsAlgPart>()
        for (i in 0 until itte.size) {
            algParts1.add(itte[i])
        }
        return Algorithm(algMarker, representation, algParts1)
    }

    fun simpleVerbatimAlgorithm(algMarker: String, vararg sayThis: String): Algorithm {
        // returns alg that says the word string (sayThis)
        return customizedVerbatimGorithm(algMarker, APVerbatim(*sayThis))
    }

    fun simpleCloudiandVerbatimAlgorithm(cldBool: CldBool, algMarker: String, vararg sayThis: String): Algorithm {
        // returns alg that says the word string (sayThis)
        return customizedVerbatimGorithm(algMarker, APCldVerbatim(cldBool!!, *sayThis))
    }

    fun strContainsList(str1: String, items: ArrayList<String>): String {
        // returns the 1st match between words in a string and values in a list.
        for (temp in items) {
            if (str1.contains(temp)) {
                return temp
            }
        }
        return ""
    }
}